File: COMPOSICION-AGREGACION/Ejercicio 1/codigo/codigo.java
//1.Sean las siguientes clases:Habitación<nombre, tamaño (en metros cuadrados)
//Métodos: mostrar_info() (muestra el nombre y tamaño de la habitación)
//Casa<dirección, habitaciones (lista de objetos de tipo Habitación) Métodos: agregar_habitacion(habitacion), mostrar_casa() (muestra la dirección y la información de todas las habitaciones)
//  a)Implementa las clases con sus constructores, getters y setters.
//  b)Crea una casa y agrega varias habitaciones.
//  c)Muestra la información de la casa y sus habitaciones.
import java.util.ArrayList;
public class Habitacion {
    private String nombre;
    private double tamano;

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m2");
    }
}

class Casa {
    private String direccion;
    private ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("Dirección: " + direccion);
        for (Habitacion h : habitaciones) {
            h.mostrarInfo();
        }
    }

    public static void main(String[] args) {
        Casa casa = new Casa("Av. Siempre Viva 742");
        casa.agregarHabitacion(new Habitacion("Sala", 20));
        casa.agregarHabitacion(new Habitacion("Cocina", 12));
        casa.mostrarCasa();
    }
}

File: COMPOSICION-AGREGACION/Ejercicio 7/codigo/codigo.py
# 7.Crea un POO para una universidad y sus estudiantes. La universidad contiene estudiantes, pero los estudiantes pueden existir independientemente de la universidad.
# Estudiante< nombre, carrera, semestre>
# Métodos: mostrar_info() (muestra el nombre, carrera y semestre del estudiante)
# Universidad<nombre, estudiantes (lista de objetos de tipo Estudiante)>
# Métodos: agregar_estudiante(estudiante), mostrar_universidad() (muestra el nombre de la universidad y la información de todos los estudiantes)
#   a)Implementa las clases con sus constructores, getters y setters.
#   b)Crea una universidad y agrega varios estudiantes.
#   c)Muestra la información de la universidad y sus estudiantes.
class Estudiante:
    def __init__(self, nombre, carrera, semestre):
        self.nombre = nombre
        self.carrera = carrera
        self.semestre = semestre

    def mostrar_info(self):
        print(f"{self.nombre} - {self.carrera} - Semestre: {self.semestre}")

class Universidad:
    def __init__(self, nombre):
        self.nombre = nombre
        self.estudiantes = []

    def agregar_estudiante(self, estudiante):
        self.estudiantes.append(estudiante)

    def mostrar_universidad(self):
        print(f"Universidad: {self.nombre}")
        for e in self.estudiantes:
            e.mostrar_info()

u = Universidad("UMSA")
u.agregar_estudiante(Estudiante("Ana", "Informática", 3))
u.agregar_estudiante(Estudiante("Carlos", "Matemáticas", 5))
u.mostrar_universidad()
